lib/stringify.js

```diff
@@ -816,27 +881,19 @@ compilerPrototype.heading = function (node) {
  * @param {Object} node - `text` node.
  * @return {string} - Raw markdown text.
  */
-compilerPrototype.text = function (node) {
-    return this.encode(node.value, node);
-};
+compilerPrototype.text = function (node, parent) {
+    var value = this.encode(node.value, node);
 
-/**
- * Stringify escaped text.
- *
- * @example
- *   var compiler = new Compiler();
- *
- *   compiler.escape({
- *     type: 'escape',
- *     value: '\n'
- *   });
- *   // '\\\n'
- *
- * @param {Object} node - `escape` node.
- * @return {string} - Markdown escape.
- */
-compilerPrototype.escape = function (node) {
-    return '\\' + node.value;
+    if (
+        value[value.length - 1] == '\\' && parent &&
+        parent.type == 'linkReference' &&
+        parent.referenceType == 'shortcut'
+    ) {
+        // Don't escape trailing backslash in shortcut links.
```

Comment:
This is here for [links-reference-style](https://github.com/wooorm/mdast/blob/b28a45ac3d8291ef2fb3e173468ac992f8c56f1f/test/tree/links-reference-style.json#L1702-L1734) test, but I suspect the test is wrong (there shouldn't be a `linkReference` node).